website/docs/r/vapp_network.html.markdown

```diff
@@ -49,15 +48,20 @@ The following arguments are supported:
   connected as sysadmin working across different organisations.
 * `vdc` - (Optional; *v2.0+*) The name of VDC to use, optional if defined at provider level.
 * `name` - (Required) A unique name for the network.
-* `vapp_name` - (Required) The vApp this VM should belong to.
+* `description` - (Optional; *v2.7+*) Description of vApp network
+* `vapp_name` - (Required) The vApp this network belongs to.
 * `netmask` - (Optional) The netmask for the new network. Default is `255.255.255.0`.
 * `gateway` (Optional) The gateway for this network.
```

Comment:
Something is not right here. So, after reading the new changelog message:
> DEPRECATIONS:
> * `resource/vcd_vapp_vm` `network.name` deprecated automatic attaching of vApp Org network when `network.type=org` and it doesn't exist. Requires to create/attach vApp network with **`vcd_vapp_network`** or `vcd_vapp_org_network` before referencing it.  

 I interpret that I should be able to create the needed vApp network not only with `vcd_vapp_org_network`, but also with `vcd_vapp_network`. So, I'm trying this:

```
resource "vcd_vapp_network" "demo_vapp_net_w_org" {
  name = "demo-vapp-net-w-org"
  vapp_name         = vcd_vapp.demo_vapp.name
  org_network_name  = vcd_network_routed.demo_routed_net_multi_pool.name
}
```

and then in VM:

```
  network {
    type               = "org"
    # 2.7.0 will throw a warning if vApp Network doesn't exist and it auto-creates it
    name = vcd_vapp_network.demo_vapp_net_w_org.org_network_name
    ip_allocation_mode = "DHCP"
  }
```

However, it says that `gateway` is required (which would conflict with the highlighted doc line of this comment):

```
$ terraform apply

Error: Missing required argument

  on xendi.tf line 320, in resource "vcd_vapp_network" "demo_vapp_net_w_org":
 320: resource "vcd_vapp_network" "demo_vapp_net_w_org" {

The argument "gateway" is required, but no definition was found.
```

So, I add the whole network definition:

```
resource "vcd_vapp_network" "demo_vapp_net_w_org" {
  name = "demo-vapp-net-w-org"
  vapp_name         = vcd_vapp.demo_vapp.name
  org_network_name  = vcd_network_routed.demo_routed_net_multi_pool.name

  gateway            = "192.168.9.1"
  netmask            = "255.255.255.0"
  dns1               = "192.168.9.1"
  dns2               = "192.168.9.2"
  dns_suffix         = "mybiz.biz"
  guest_vlan_allowed = true

  # 2.7.0
  #org_network_name = vcd_network_routed.demo_routed_net_multi_pool.name

  static_ip_pool {
    start_address = "192.168.9.51"
    end_address   = "192.168.9.100"
  }

  dhcp_pool {
    start_address = "192.168.9.2"
    end_address   = "192.168.9.50"
  }
}
```

And run apply, but then I get the DEPRECATED message even though I did create the `vcd_vapp_network` as per your changelog message:

```
vcd_vapp_network.demo_vapp_net_w_org: Creating...
vcd_vapp_network.demo_vapp_net_w_org: Still creating... [10s elapsed]
vcd_vapp_network.demo_vapp_net_w_org: Still creating... [20s elapsed]
vcd_vapp_network.demo_vapp_net_w_org: Creation complete after 25s [id=b68affa0-c540-46c9-a2ec-0180041c8973]
vcd_vapp_vm.demo_vm_wordpress: Modifying... [id=urn:vcloud:vm:ef2bb818-6462-4db5-bd71-0df42cdec2ad]
vcd_vapp_vm.demo_vm_wordpress: Still modifying... [id=urn:vcloud:vm:ef2bb818-6462-4db5-bd71-0df42cdec2ad, 10s elapsed]
DEPRECATED: attaching an Org network `demo-net-multi_pool` to a vApp `demo-web` through VM's network block alone is deprecated. Network should be first attached to a vApp by creating a `vcd_vapp_org_network` resource and only then referenced in the network block.
```

Finally, the result of such config is that I get **two** networks in the vApp! My expected one "demo-vapp-net-w-org " from `vcd_vapp_network`, but also one "demo-net-multi_pool" auto-created by the VM: 

![Screenshot 2020-03-05 at 09 15 06](https://user-images.githubusercontent.com/11330689/75957009-d0066300-5ec1-11ea-944d-8019bd0fbe91.png)

And, the VM did **not** connect to the "demo_vapp_net_w_org":

![Screenshot 2020-03-05 at 09 19 28](https://user-images.githubusercontent.com/11330689/75957343-823e2a80-5ec2-11ea-9ff6-d4d1867f1040.png)

I believe the changelog message is incorrect and should only reference the `vcd_vapp_org_network`, but I will leave it up to you to confirm.

CC: @Didainius

Reply:
yes, only `vcd_vapp_org_network`